clippy_lints/src/methods/mod.rs

```diff
@@ -1289,46 +1281,37 @@ fn lint_or_fun_call<'a, 'tcx>(
             (&paths::RESULT, true, &["or", "unwrap_or"], "else"),
         ];
 
-        // early check if the name is one we care about
-        if know_types.iter().all(|k| !k.2.contains(&name)) {
-            return;
-        }
+        if_chain! {
+            if know_types.iter().any(|k| k.2.contains(&name));
 
-        let mut finder = FunCallFinder { cx: &cx, found: false };
-        finder.visit_expr(&arg);
-        if !finder.found {
-            return;
-        }
+            let mut finder = FunCallFinder { cx: &cx, found: false };
+            if { finder.visit_expr(&arg); finder.found };
```

Comment:
This should have been flagged by `block_in_if_condition_stmt`

Reply:
This may have been skipped because of a macro check in the lint?